symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -0,0 +1,138 @@
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() throws Throwable {
+        this.datafeedId = this.retrieveDatafeedIdFromDisk();
+
+        if (this.datafeedId == null) {
+            this.datafeedId = this.createDatafeedAndSaveToDisk();
+        }
+
+        log.debug("Start reading events from datafeed {}", datafeedId);
+        do {
+            this.started.set(true);
+            this.readDatafeed();
+        } while (this.started.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    private void readDatafeed() throws Throwable {
+        this.retry.executeCheckedSupplier(() -> {
+            try {
+                List<V4Event> events = datafeedApi.v4DatafeedIdReadGet(datafeedId, authSession.getSessionToken(), authSession.getKeyManagerToken(), null);
+                if (events != null && !events.isEmpty()) {
+                    handleV4EventList(events);
+                }
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    authSession.refresh();
+                }
+                if (e.isClientError()) {
+                    log.info("Recreate a new datafeed and try again");
+                    datafeedId = this.createDatafeedAndSaveToDisk();
+                }
+                throw e;
+            }
+            return null;
+        });
+    }
+
+    private String createDatafeedAndSaveToDisk() throws Throwable {
+        log.debug("Start creating a new datafeed and save to disk");
+        return this.retry.executeCheckedSupplier(() -> {
+            try {
+                String id = this.datafeedApi.v4DatafeedCreatePost(authSession.getSessionToken(), authSession.getKeyManagerToken()).getId();
+                this.writeDatafeedIdToDisk(id);
+                log.debug("Datafeed: {} was created and saved to disk", id);
+                return id;
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    authSession.refresh();
+                }
+                throw e;
+            }
+        });
+    }
+
+    private String retrieveDatafeedIdFromDisk() {
+        log.debug("Start retrieving datafeed id from disk");
+        String datafeedId = null;
+        try {
+            File file = this.getDatafeedIdFile();
+            Path datafeedIdPath = Paths.get(file.getPath());
+            String[] persistedDatafeed = Files.readAllLines(datafeedIdPath).get(0).split("@");
+            datafeedId = persistedDatafeed[0];
```

Comment:
what is the file is empty or has been changed manually?
We should probably just ignore it silently (here it will break with an exception)